### PR TITLE
Batch builder for transferTokens method with the unit test.

### PR DESCRIPTION
**Pull request by alex-kooper**

### CODE REVIEW

File: packages/fa2-interfaces/__tests__/transfer-batch.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+describe('Test building a batch of transfers', () => {
+  test('mergeing source correctly', () => {
+    const transfers = transferBatch()
+      .withTransfer('tzFrom1', 'tzTo1', 1, 1)
+      .withTransfer('tzFrom1', 'tzTo2', 2, 1)
+      .withTransfer('tzFrom2', 'tzTo1', 3, 1)
+      .withTransfer('tzFrom1', 'tzTo3', 3, 1).transfers;
+
+    expect(transfers).toHaveLength(3);
```
Comment (emishur) [inline packages/fa2-interfaces/__tests__/transfer-batch.ts:11]: Check `from_` parameter as well. 
Also do an exhaustive check of the whole  batch.
Also add a test sequence for from1, from2, from1.

File: packages/fa2-interfaces/__tests__/transfer-batch.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { transferBatch } from '../src/transfer-batch';
```
Comment (emishur) [inline packages/fa2-interfaces/__tests__/transfer-batch.ts:1]: Follow test files naming convention. Test file names should end with `.test.ts` suffix

File: packages/fa2-interfaces/src/transfer-batch.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ * It merges the subsequent transfers with the same source (from_ field)
+ * 
+ * Usage example:
+ * ```typescript
+ * const transfers = transferBatch()
+ *   .withTransfer('tzFromAccount1', 'tzToAccount1', 1, 1)
+ *   .withTransfer('tzFromAccount1', 'tzToAccount2', 2, 1)
+ *   .transfers;
+ *
+ * contract.transferTokens(transfers)
```
Comment (emishur) [inline packages/fa2-interfaces/src/transfer-batch.ts:15]: Consider using batch api in transfer implementation in `tznft` `contracts.ts`

File: packages/fa2-interfaces/src/transfer-batch.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+ * @param transfers a list of transfers to add more transfers to 
+ * (normally use the default empty list)
+ * 
+ * @returns a batch of transfers that can be populated with transfers
+ */
+export const transferBatch = (transfers: Transfer[] = []) => ({
+  transfers,
+
+  withTransfer: (from: address, to: address, tokenId: nat, amount: nat) => {
+    const last = transfers[transfers.length - 1];
```
Comment (emishur) [inline packages/fa2-interfaces/src/transfer-batch.ts:27]: How would last work with an empty transfer list?
(Judging by the following code, it should work fine…)

File: packages/fa2-interfaces/__tests__/transfer-batch.ts (hunk 1 above)
Comment (alex-kooper) [reply packages/fa2-interfaces/__tests__/transfer-batch.ts]: Done

File: packages/fa2-interfaces/__tests__/transfer-batch.ts (hunk 2 above)
Comment (alex-kooper) [reply packages/fa2-interfaces/__tests__/transfer-batch.ts]: Done

File: packages/fa2-interfaces/src/transfer-batch.ts (hunk 3 above)
Comment (alex-kooper) [reply packages/fa2-interfaces/src/transfer-batch.ts:15]: Done

File: packages/fa2-interfaces/src/transfer-batch.ts (hunk 4 above)
Comment (alex-kooper) [reply packages/fa2-interfaces/src/transfer-batch.ts:27]: last will be undefined in that case. It is all tested and works.

File: packages/fa2-interfaces/src/fa2-interface.ts [hunk 5]
```diff
@@ -77,34 +77,53 @@ export interface Transfer {
 }
 
 /**
- * Operator update parameter
+ * Operator update parameters
  */
-export interface OperatorUpdate {
+export interface OperatorUpdateParams {
```
Comment (emishur) [inline packages/fa2-interfaces/src/fa2-interface.ts:82]: Shouldn't it be a singular form `OperatorUpdateParam`?

Comment (alex-kooper) [reply packages/fa2-interfaces/src/fa2-interface.ts:82]: I think it should be plural. I am looking at your own comment:
```
/**
 * Operator update parameters
 */
export interface OperatorUpdateParams {
```
???

Comment (alex-kooper) [reply packages/fa2-interfaces/src/fa2-interface.ts:82]: `Operator update parameters` comment was written by YOU.

**Comment by emishur** [comment]
Approved

**Review by emishur — APPROVED** [review]

**Pull request merged**